src/main/java/apoc/convert/Convert.java

```diff
@@ -163,5 +165,85 @@ else if (list.getClass().isArray() && !list.getClass().getComponentType().isPrim
 	public List<Relationship> toRelationshipList(@Name("list") Object list) {
         return convertToList(list, Relationship.class);
 	}
-	
+
+    @UserFunction
+    @Description("apoc.convert.toInteger(value) | tries it's best to convert the value to an integer")
+    public Number toInteger(@Name("object") Object obj) {
+        if (obj == null || obj.equals("")) {
+            return null;
+        }
+
+        Types varType = Types.of(obj);
+        Number result;
+        switch (varType) {
+            case INTEGER:
+            case FLOAT:
+                result = ((Number) obj).intValue();
```

Comment:
Cypher only knows long (which it calls Integer), so we should convert it to Long (longValue())